File: bigtangle-typescript/src/net/bigtangle/params/ReqCmd.ts
// Imports for context (not used directly in enum, but for consistency with project structure)
import '../core';
import '../utils';
import '../exception';
import '../params';
import '../script';

export enum ReqCmd {
    // Block
    saveBlock = "saveBlock",
    getBlockByHash = "getBlockByHash",
    findBlockEvaluation = "findBlockEvaluation",
    searchBlockByBlockHashs = "searchBlockByBlockHashs",
    batchBlock = "batchBlock",
    blockEvaluationFromHashs = "blockEvaluationFromHashs",
    getTip = "getTip",
    adjustHeight = "adjustHeight",
    findRetryBlocks = "findRetryBlocks",
    // Chain
    getChainNumber = "getChainNumber",
    getAllConfirmedReward = "getAllConfirmedReward",
    blocksFromChainLength = "blocksFromChainLength",
    blocksFromNonChainHeight = "blocksFromNonChainHeight",
    // Token
    searchTokens = "searchTokens",
    getTokenById = "getTokenById",
    getTokenIndex = "getTokenIndex",
    getTokenSignByAddress = "getTokenSignByAddress",
    searchExchangeTokens = "searchExchangeTokens",
    searchTokenDomain = "searchTokenDomain",
    searchWebTokens = "searchWebTokens",
    searchContractTokens = "searchContractTokens",
    getTokenSignByTokenid = "getTokenSignByTokenid",
    signToken = "signToken",
    getTokenSigns = "getTokenSigns",
    getTokenPermissionedAddresses = "getTokenPermissionedAddresses",
    getDomainNameBlockHash = "getDomainNameBlockHash",
    // Block Order
    getOrders = "getOrders",
    getOrdersTicker = "getOrdersTicker",
    // Outputs
    getOutputByKey = "getOutputByKey",
    getOutputs = "getOutputs",
    getOutputsHistory = "getOutputsHistory",
    outputsOfTokenid = "outputsOfTokenid",
    getBalances = "getBalances",
    outputsByBlockhash = "outputsByBlockhash",
    getAccountBalances = "getAccountBalances",
    // Payment
    launchPayMultiSign = "launchPayMultiSign",
    payMultiSign = "payMultiSign",
    getPayMultiSignList = "getPayMultiSignList",
    getPayMultiSignAddressList = "getPayMultiSignAddressList",
    payMultiSignDetails = "payMultiSignDetails",
    // User data
    getUserData = "getUserData",
    userDataList = "userDataList",
    // Subtangle
    regSubtangle = "regSubtangle",
    updateSubtangle = "updateSubtangle",
    getSessionRandomNum = "getSessionRandomNum",
    // Permissioned
    addAccessGrant = "addAccessGrant",
    deleteAccessGrant = "deleteAccessGrant",
    // Check point value
    getCheckPoint = "getCheckPoint",
    serverinfolist = "serverinfolist"
}

File: bigtangle-core/src/main/java/net/bigtangle/params/ReqCmd.java
/*******************************************************************************
 *  Copyright   2018  Inasset GmbH. 
 *  
 *******************************************************************************/
package net.bigtangle.params;

public enum ReqCmd {
	// Block
	submitTransaction, submitTransactions, getBlockByHash, findBlockEvaluation, searchBlockByBlockHashs, batchBlock,
	// Bridge
	processPegIn, processPegOut,

	getTip, getTips, adjustHeight, findRetryBlocks,
	// Chain
	getChainNumber, getAllConfirmedReward, blocksFromChainLength,blocksFromNonChainHeight,
	// Mempool
	getPendingTransactions,
	// Token
	searchTokens, getTokenById, getTokenIndex, getTokenSignByAddress, searchExchangeTokens, searchWebTokens,

	getTokenSignByTokenid, signToken, getTokenSigns, getTokenPermissionedAddresses, getDomainNameBlockHash,
	// Block Order
	getOrders, getOrdersTicker,
	// Outputs
	getOutputByKey, getOutputs, getOutputsHistory, outputsOfTokenid, getBalances,getAccountBalances, getTransactionStatus, getTransactionsStatusByAddress,

	// payment
	launchPayMultiSign, payMultiSign,

	getPayMultiSignList, getPayMultiSignAddressList, payMultiSignDetails,
	 
	//user data
	getUserData, userDataList,  
	// subtangle
	regSubtangle, updateSubtangle, getSessionRandomNum,
	// permissioned
	addAccessGrant, deleteAccessGrant,
    // check point value
    getAnchors,
    // PoS
    submitAttestation, getAttestations, processWithdrawal, submitSlashingProof, requestValidatorExit,
    stakeDeposit, activateValidator, getValidators, getBaseFee, setValidatorKey, getValidatorKey,
    // NFT
    createNft, saveUserdata;
}
